frame/staking/src/lib.rs

```diff
@@ -2652,6 +2731,8 @@ impl<T: Trait> Module<T> {
 					distribution[0].1 == OffchainAccuracy::one(),
 					Error::<T>::PhragmenBogusSelfVote,
 				);
+				// all good. We can refund some weight now.
```

Comment:
I'm not sure I understand this comment, this refund will only happen after build_support (i.e. the check that nomination actually nominate winners)

Also we should refund way more no ? not `original_weight-db::reads(winners.len())` but `original_weight - db::reads(edge_count) + db::reads(winners.len())`

Reply:
So this is trying to refund the slashingSpan and yeah it is wrong. Let me try and write it down here and now: 

We charge `compact.edge_count() * read(1)` originally for slashingSpan. If solution is invalid, this is okay since this is the worse case. 

If solution was valid, we know that a valid solution read at most the slashingSpan of all winners, hence we must reduce it to `winner.len() * read(1)`. 

I will update it like this.

Reply:
the update looks good